### DISCUSSION TITLE
Feature Request: Track landmarks (tensor of coordinates)

### DISCUSSION THREAD

**Discussion by Schobs** [TorchIO-project/torchio#1003] [Ideas]
Is it possible to augment landmarks along with segmentation maps? This would be a tensor of 2D or 3D coordinates relating to anatomical landmarks. Could you add a feature to track them through the spatial augmentations, or is there a way of retrieving the matrices of the transforms to apply manually? At the moment I use imgaug, but torch.io would be better.

Finally, how much more efficient would this be compared to simply creating a segmentation map per landmark, with the landmark of interest as the only positive class in the label map?

Thank you!

**Comment by Schobs** [comment]
@fepegar hey fernando, I'd be happy to branch and have a go at this. 

If convenient, please could you or a colleague give me a pointer on where to start with this? Thank you.

**Comment by romainVala** [comment]
Hello, 
this will be not that easy to implement: one should add a new datatype (in addition to `ScalarImage` and `LabelMap`, one should add a LandmarkImage. ) Then one should consider this type of data for affine and elastics deformation and thus a well understanding (and experient) will be needed to use the proper itk function to apply those deformation to point (instead) of images

A much straitforward solution would be (as you mentioned) to transform your landmark to a binary mask and use it as a `LabelMap` .. how much efficient will it be ?

well, we never sure until we test the 2 solutions, but I guess it will only add uncertainty up to the voxel resolution you have. So depending of your global problem (and the exact shape of the landmark) it might not be a big issue to approximate the landmark to a voxel  (you get a precision with the magnitude order of the voxel resolution)

I hope it helps

cheers
